CMakeLists.txt

```diff
@@ -69,7 +69,7 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   set(C_WARNING_ARGS "${C_WARNING_ARGS} -Wno-c++98-compat -Wno-padded") # these are annoying
   set(C_WARNING_ARGS "${C_WARNING_ARGS} -Wno-documentation") # gtk-doc != doxygen
 else()
-  set(C_WARNING_ARGS "${C_WARNING_ARGS} -Wall -Wextra -Wpedantic -Wformat=2")
+  set(C_WARNING_ARGS "${C_WARNING_ARGS} -Wall -Wformat=2") # Use -Wextra and -Wpedantic explicitly if needed
```

Comment:
Why did you remove these two. I find it actually quite useful seeing as many warnings as possible during nightly builds.

Reply:
They spit out a whole lot of warnings that cannot and need not be really fixed (for properly done type casting, etc). I think we should bear in mind that our primary build audience are users of all sorts of distros needing to build these components for themselves. It makes sense to provide them with a nice clean build experience, and explicitly enable overkill warning mode for development builds.